Term entry] UI and ux design UI ux (ideation) user pain points (#4006)
Codecademy/docs: content/uiux/user-pain-points/user-pain-points.md

---
Title: 'User Pain Points'
Description: 'Refers to issues a user incounters when using a program or website.'
Subject:
 - 'Developer Tools'
 - 'Web Design'
 - 'Web Development'
Tags:
 - 'Best Practices'
 - 'Developement'
 - 'UI'
 - 'UX'
ContentCatalog:
- 'intro-to-ui-ux'
- 'paths/front-end-engineer-career-path'
---

> **User pain points** are areas of difficulty or frustration a user encounters when using a website or program that make the experience undesirable. User pain points can be minor inconveinences to major issues preventing users from accessing the product or service entirely. Identifying and resolving these issues is a crucial part of web development and should be considered throughout any project. Pain points are commonly described in three levels with four types at each level

Reviewer: ```suggestion
**User Pain Points** are areas of difficulty or frustration for a user when using a product that makes the experience undesirable. These can be minor inconveniences or issues preventing users from accessing the product or service entirely.  User Pain Points can also be difficulties users face with their daily tasks. Such problems are identified at the beginning of the product development process, to build user-centric solutions. They can be identified through User Research within the User Experience (UX) design process.

User Pain Points within existing applications are commonly described in three levels with four types at each level given below.
```

> **User pain points** are areas of difficulty or frustration a user encounters when using a website or program that make the experience undesirable. User pain points can be minor inconveinences to major issues preventing users from accessing the product or service entirely. Identifying and resolving these issues is a crucial part of web development and should be considered throughout any project. Pain points are commonly described in three levels with fours types at each level

Reviewer 2: ```suggestion
**User pain points** are areas of difficulty or frustration a user encounters when using a website or program that make the experience undesirable. User pain points can be minor inconveniences to major issues preventing users from accessing the product or service entirely. Identifying and resolving these issues is a crucial part of web development and should be considered throughout any project. Pain points are commonly described in three levels with four types at each level
```

## Pain Point Levels

>  - Interaction: users interaction with product or support
>  - Journey: the steps it takes a user to complete a task
>  - Relationship: users lifetime experience with a product or brand 

Reviewer 2: ```suggestion
- `Interaction`: User's interaction with product or support
- `Journey`: The steps it takes a user to complete a task
- `Relationship`: User's lifetime experience with a product or brand 
```

## Pain Point Types

>  - Financial: relates to paywalls and subscription services that locks a user out.
>  - Product: relates to the performance, quality and usability issues while a using a product
>  - Process: relates to the steps required to complete a task
>  - Support: relates to how organizations respond to user questions

Reviewer 2: ```suggestion
- `Financial`: relates to paywalls and subscription services that lock a user out.
- `Product`: relates to the performance, quality, and usability issues while using a product
- `Process`: relates to the steps required to complete a task
- `Support`: relates to how organizations respond to user questions
```

 
## Example

> In the example below there are many elements of the web page that are sure to cause frustration for any user. A user might find it difficult to differentiate between what is an item for sale and what is an advertisment. There are also many accessibility issues and following [Accessibility Best Practices](https://www.w3.org/WAI/tips/designing/Accessibility) will aid in solving many common issues.

Reviewer 2: ```suggestion
In the example below there are many elements of the web page that are sure to cause frustration for any user. A user might find it difficult to differentiate between what is an item for sale and what is an advertisement. There are also many accessibility issues and following [Accessibility Best Practices](https://www.w3.org/WAI/tips/designing/Accessibility) will aid in solving many common issues.
```

> ![Example of a crowded website layout](https://raw.githubusercontent.com/Codecademy/docs/main/media/user-pain-points-example.png)

Reviewer: This example seems a little vague. Can you use a simpler example to explain a user pain point? It isn't necessary to use an image. 

If you have trouble with this, let me know.
